sycl/test-e2e/Basic/built-ins/marray_common.cpp

```diff
@@ -0,0 +1,122 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#ifdef _WIN32
+#define _USE_MATH_DEFINES // To use math constants
+#include <cmath>
+#endif
+
+#include <sycl/sycl.hpp>
+
+#define TEST(FUNC, MARRAY_ELEM_TYPE, DIM, EXPECTED, DELTA, ...)                \
```

Comment:
Do we have to use a macro here? Seems like we can get away with a function template, which would be preferrable. The same goes for the other test file.

Reply:
I would suggest to implement this in a separate PR for all marray builtin tests, including these.